### ISSUE TITLE
[Accessibility]: Compare page: Diff overview is difficult to undestand

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35535]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34426

### Problem description

<img width="431" alt="image" src="https://user-images.githubusercontent.com/9516420/168786118-37ff69c4-a14e-40c3-97f1-872425cb7c47.png">

The colored dots aren't red out by the screen reader, which would be fine - except that only "28" is read out with no real context.

### Expected behavior

We should instead provide a text overview of the colored dots (e.g. 28 lines changed, 20 additions, 8 deletions)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
